modules/imgproc/src/smooth.dispatch.cpp

```diff
@@ -644,13 +644,31 @@ void GaussianBlur(InputArray _src, OutputArray _dst, Size ksize,
     CV_UNUSED(useOpenCL);
 
     int sdepth = CV_MAT_DEPTH(type), cn = CV_MAT_CN(type);
+    bool OpenCL_applicable = sdepth == CV_8U &&
+            ((ksize.width == 3 && ksize.height == 3) ||
+            (ksize.width == 5 && ksize.height == 5));
+
+    static bool param_check_gaussian_blur_bitexact_kernels = utils::getConfigurationParameterBool("OPENCV_GAUSSIANBLUR_CHECK_BITEXACT_KERNELS", false);
+
+    if (!(useOpenCL && OpenCL_applicable) || !param_check_gaussian_blur_bitexact_kernels)
+    {
+        Mat src = _src.getMat();
+        Mat dst = _dst.getMat();
+
+        Point ofs;
+        Size wsz(src.cols, src.rows);
+        if(!(borderType & BORDER_ISOLATED))
+            src.locateROI( wsz, ofs );
+
+        CALL_HAL(gaussianBlur, cv_hal_gaussianBlur, src.ptr(), src.step, dst.ptr(), dst.step, src.cols, src.rows, sdepth, cn,
```

Comment:
This API is for pre-bitexact implementation.

Bitexact path should has own (new!) HAL API function.

(we don't change semantic existed API, we add new API with new requirements)